From d46e2199ec1fb8e1a0ed0793f45afcbe66e5e45c Mon Sep 17 00:00:00 2001
From: Rohit Sharma <rohit.sharma@r2p.com>
Date: Thu, 16 Feb 2023 07:49:10 +0100
Subject: [PATCH 3/3] Balena env on uboot

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Rohit Sharma <rohitsharmanotify@gmail.com>
---
 include/env_default.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/env_default.h b/include/env_default.h
index c0df39d62f..b67a328d2d 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -9,6 +9,7 @@
 
 #include <env_callback.h>
 #include <linux/stringify.h>
+#include <env_resin.h>
 
 #ifndef USE_HOSTCC
 #include <generated/environment.h>
@@ -28,6 +29,7 @@ char default_environment[] = {
 #else
 const char default_environment[] = {
 #endif
+	BALENA_ENV
 #ifndef CONFIG_USE_DEFAULT_ENV_FILE
 #ifdef	CONFIG_ENV_CALLBACK_LIST_DEFAULT
 	ENV_CALLBACK_VAR "=" CONFIG_ENV_CALLBACK_LIST_DEFAULT "\0"
-- 
2.39.2

